Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -222,71 +95,54 @@ - (void)commontInit
-    [self.footerView autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:self.payloadView];
-
-    [self.mediaMaskingView autoPinEdgeToSuperviewEdge:ALEdgeTop];
-
-    [self.textBubbleImageView autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:self.mediaMaskingView];
-    [self.textBubbleImageView autoPinEdgeToSuperviewEdge:ALEdgeBottom];
+    [self.bubbleView autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:self.dateHeaderLabel];
+    [self.footerView autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:self.bubbleView];
 
     [self.footerView autoPinEdgeToSuperviewEdge:ALEdgeBottom];
     [self.footerView autoPinWidthToSuperview];
 
-    UITapGestureRecognizer *mediaTap =
-        [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleMediaTapGesture:)];
-    [self.mediaMaskingView addGestureRecognizer:mediaTap];
-
-    UITapGestureRecognizer *textTap =
-        [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleTextTapGesture:)];
-    [self.textBubbleImageView addGestureRecognizer:textTap];
+    self.contentView.userInteractionEnabled = YES;
 
-    UILongPressGestureRecognizer *mediaLongPress =
-        [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleMediaLongPressGesture:)];
-    [self.mediaMaskingView addGestureRecognizer:mediaLongPress];
+    UITapGestureRecognizer *tap =
+        [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleTapGesture:)];
+    [self.contentView addGestureRecognizer:tap];
 
-    UILongPressGestureRecognizer *textLongPress =
-        [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleTextLongPressGesture:)];
-    [self.textBubbleImageView addGestureRecognizer:textLongPress];
+    UILongPressGestureRecognizer *longPress =
+        [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleLongPressGesture:)];
+    [self.contentView addGestureRecognizer:longPress];
 
-    PanDirectionGestureRecognizer *panGesture =
-        [[PanDirectionGestureRecognizer alloc] initWithDirection:(self.isRTL ? PanDirectionLeft : PanDirectionRight)
-                                                          target:self
-                                                          action:@selector(handlePanGesture:)];
+    PanDirectionGestureRecognizer *panGesture = [[PanDirectionGestureRecognizer alloc]
+        initWithDirection:(self.isRTL ? PanDirectionLeft : PanDirectionRight)target:self
+                   action:@selector(handlePanGesture:)];
     [self addGestureRecognizer:panGesture];
 }
 
+- (OWSMessageTextView *)newTextView
+{
+    OWSMessageTextView *textView = [OWSMessageTextView new];
+    textView.backgroundColor = [UIColor clearColor];
+    textView.opaque = NO;
+    textView.editable = NO;
+    textView.selectable = YES;
+    textView.textContainerInset = UIEdgeInsetsZero;
+    textView.contentInset = UIEdgeInsetsZero;
+    textView.textContainer.lineFragmentPadding = 0;
```

Comment:
Here's an important fix that we may want to apply elsewhere: the text view has 3 separate properties that lead to a visual margin between the text and the view's bounds.  We need to zero them all out.